core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java

```diff
@@ -384,7 +384,11 @@ private void onShardOperation(final NodeRequest request, final Object[] shardRes
                 e.setIndex(shardRouting.getIndex());
                 e.setShard(shardRouting.shardId());
                 shardResults[shardIndex] = e;
-                logger.debug("[{}] failed to execute operation for shard [{}]", e, actionName, shardRouting.shortSummary());
+                if (logger.isDebugEnabled() && !TransportActions.isShardNotAvailableException(e)) {
```

Comment:
> e != t :)

@bleskes Good eye!

> I think this will be simpler to read if it has a top level with TransportActions.isShardNotAvailableException(t) and then the right handling bellow it.

I'm not so sure. Unless you had something different in mind than

```diff
diff --git a/core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java b/core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java
index f2092e3..c32de70 100644
--- a/core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java
+++ b/core/src/main/java/org/elasticsearch/action/support/broadcast/node/TransportBroadcastByNodeAction.java
@@ -384,10 +384,14 @@ public abstract class TransportBroadcastByNodeAction<Request extends BroadcastRe
                 e.setIndex(shardRouting.getIndex());
                 e.setShard(shardRouting.shardId());
                 shardResults[shardIndex] = e;
-                if (logger.isDebugEnabled() && !TransportActions.isShardNotAvailableException(e)) {
-                    logger.debug("[{}] failed to execute operation for shard [{}]", e, actionName, shardRouting.shortSummary());
-                } else if (logger.isTraceEnabled() && TransportActions.isShardNotAvailableException(t)) {
-                    logger.trace("[{}] failed to execute operation for shard [{}]", e, actionName, shardRouting.shortSummary());
+                if (TransportActions.isShardNotAvailableException(t)) {
+                    if (logger.isTraceEnabled()) {
+                        logger.trace("[{}] failed to execute operation for shard [{}]", t, actionName, shardRouting.shortSummary());
+                    }
+                } else {
+                    if (logger.isDebugEnabled()) {
+                        logger.debug("[{}] failed to execute operation for shard [{}]", t, actionName, shardRouting.shortSummary());
+                    }
                 }
             }
         }
```

it's just more lines of code to say the same thing.